### ISSUE TITLE
Deliver Jakarta Connectors Specification Version update for Jakarta EE 9

### ISSUE THREAD

**Issue by kwsutter** [eclipse-ee4j/jca-api#60]
**Is your feature request related to a problem? Please describe.**
Jakarta EE 9 is the next major release, with the main focus of moving from the ```javax``` namespace to the ```jakarta``` namespace.

**Describe the solution you'd like**
This issue will be used to track the progress of this work effort via the [Jakarta EE 9 Project board](https://github.com/orgs/eclipse-ee4j/projects/17).

**Additional context**
[Jakarta EE Specification Process](https://jakarta.ee/about/jesp/)
[Eclipse Project Handbook](https://www.eclipse.org/projects/handbook/#release)
[Eclipse Release Record for Jakarta EE 9](https://projects.eclipse.org/projects/ee4j.jakartaee-platform/releases/9)

**ToDo**
- [x] **[Create Eclipse Release Record](https://www.eclipse.org/projects/handbook/#pmi-commands-release) in your respective Jakarta Specification Project.**
 Most Component Release Records will just reference the [Jakarta EE 9 Platform Release Plan](https://eclipse-ee4j.github.io/jakartaee-platform/jakartaee9/JakartaEE9ReleasePlan).  But, if your Component deviates at all from the Platform Release Plan, then a description of the changes will be required in the Release Record.
- [x] **Initiate a ballot for your Jakarta Release Record/Plan.**
 Again, if your component is only doing the required minimum as defined by the [Jakarta EE 9 Platform Release Plan](https://eclipse-ee4j.github.io/jakartaee-platform/jakartaee9/JakartaEE9ReleasePlan), then no separate ballot is required.  You are already approved.  But, if your component deviates from the Platform Release Plan, then you will need to initiate your own ballot as defined by the [Jakarta EE Specification Process](https://jakarta.ee/about/jesp/).
- [x] **Jakarta-ize your Specification document (if applicable)**
 Many of the Jakarta EE components now have the source for their Specification documents.  It is strongly recommended that these Specification documents are properly updated to represent Jakarta EE instead of Java EE.  Some [helpful instructions are documented](https://github.com/jakartaee/specification-committee/blob/master/steps_javaee_to_jakartaee.adoc) here.
- [x] **javax -> jakarta Spec updates**
 If your component has Specification source, then all javax package references need to be updated to use jakarta package references.
- [x] **javax -> jakarta API updates**
 Your component APIs need to move from using the javax namespace to using the jakarta namespace.
- [x] **Release Candidate (RC) of Jakarta API in Maven Central**
 A Release Candidate of your component API should be pushed to Maven Central as soon as humanly possible.  This will allow other dependent components to utilize your APIs as they are updating their APIs.  Multiple revisions of these Release Candidate APIs are allowed (and probably expected) during the development cycle.
- [x] **javax -> jakarta TCK updates**
 Your component TCK needs to be updated to use the jakarta namespace.
- [x] **javax -> jakarta Compatible Implementation updates**
 Your compatible implementation that will be used to demonstrate completeness of the Specification needs to be updated to use the jakarta namespace.
- [x] **Final Jakarta API available in Staging Repo**
 When ready, your final version of the API needs to be staged to get ready for the PR review and approvals.
- [x] **Draft Specification and Apidoc PRs ready for review**
 Like we did for Jakarta EE 8, draft PRs need to be created and reviewed in preparation for the final ballots.
- [x] **Release Review Ballot started**
 Each Jakarta EE component will need to initiate a separate Release Review ballot after proper reviewing has completed.  To be clear, there will not be a blanket Release Review for all of Jakarta EE 9 like we did for the Plan Review.  Each component needs a separate Release Review.

**Comment by kwsutter** [comment]
@smillidge Why was this Issue closed?  Are you tracking JCA (Connectors) via some other Issue with similar checkpoints?  Or, did you accidentally close it and it needs to be re-opened?  (I don't have the power to reopen in jca-api.). Thanks!

**Comment by smillidge** [comment]
No idea I just moved it to in progress on the tracking board.

**Comment by kwsutter** [comment]
Can I ask this team to please create and assign the following labels to this Issue? Thank you!
[Epic]
[wave:5]

**Comment by mcserra** [comment]
I @smillidge, can i help with "Jakarta-ize your Specification document"? Thanks

**Comment by smillidge** [comment]
Thanks the 2.0.0 branch has the spec doc from Oracle. First step I think is to split it into chapters to make the conversion easier to do in parallel.

**Comment by mcserra** [comment]
I was just seeing the document, and I have doubts of what needs to change prior to Chapter 13, so I can start from there. But the chapters are so small that maybe I could start changing from 13 to 18 or something like that (?)

**Comment by smillidge** [comment]
Don't worry too much about the early chapters. I will also review any PR

**Comment by bbulk** [comment]
Hi @smillidge 
as I mentioned, I also like to help with converting / Jakarta-izing the specification document.
Do you still plan to create separate issues or split the document as e.g. in the platform spec ?

I have some free time over the next days and would just leave a comment where I start.

**Comment by smillidge** [comment]
Hi @bbulk I realized there were loads of sections so splitting up didn't make sense. 
Just make a comment here. Looks like @mcserra is taking sections 13 - 23 so pick a bunch of sections and feel free to make a PR for those sections. My team is currently working on updating all the images so don't worry about those.

**Comment by bbulk** [comment]
Hi,
I've created a pull request for sections 24 to 36.

I noticed there is no conn.html for all the links, but I assume it will work correctly.
Also I'm not sure about how the code snippets should look like, soI just changed the package names.

**Comment by mcserra** [comment]
Hi @smillidge. I'll pick 37 to 50, ok?

**Comment by smillidge** [comment]
There's already PRs up to 94 so anything beyond that go for it

**Comment by mcserra** [comment]
Ok, 95 to 110 it is. Thanks

**Comment by mcserra** [comment]
I'll pick 111 to 150 next, if nobody is working on it. Ok?

**Comment by smillidge** [comment]
Yep go for it

**Comment by bbulk** [comment]
I'll go for 151 to 170.

**Comment by mcserra** [comment]
I'll continue with 171 to 190

**Comment by mcserra** [comment]
Hi. I was starting the migration of the next section (406 forward), but I'm having some doubts as there is an appendix with 

> Document Type Definitions (DTDs) and XML schemas (XSD) for Deployment Descriptors from
previous versions of the J2EE Connector Architecture specification

And it says that 

> All Java EE products are required to support these DTDs as well

In this appendix there are references to the javax namespace, and if I change it I guess it will be wrong, so I don't know what to do here...

**Comment by smillidge** [comment]
@mcserra I just committed the final section. I will create an issue to work out what to do with the appendices and the XSDs etc as I'm not sure either.
